datamol/viz/_viz.py

```diff
@@ -120,6 +121,13 @@ def to_image(
         else:
             _kwargs[k] = v
 
+    # Check if we are in a Jupyter notebook or IPython display context
+    # If so, conditionally add the maxMols argument
+    in_notebook = get_ipython() is not None
+
+    if in_notebook:
+        _kwargs["maxMols"] = max_mols
```

Comment:
I wonder whether we should actually always pass `max_mols` to `maxMols` in ` Draw.MolsToGridImage` below? What do you think?

Reply:
hum actually I am not sure... what is your use case here?

On one side, I would say, the user will expect something like `max_mols` to be identical to the upstream `maxMols`. But on the other side, datamol is doing some preprocessing which where we "cut" to `max_mols` earlier. 

Whatever our final decision, it should be clearly documented in the docstring!

Reply:
My first commit here passed `max_mols` to `maxMols` in `Draw.MolsToGridImage`, but all tests failed. It turns out this arg is only valid when running in an `IPython` context.

Reply:
>  but all tests failed

Can you paste the failing traceback here?

Reply:
https://github.com/datamol-io/datamol/actions/runs/7200215682/job/19613651585

Reply:
sorry I should've been more clear, all tests related to `to_image` failed.

Reply:
That sounds good to me. Might be worth adding a logger warning to alert the user that they should be using `max_mols_ipython` if they want to plot >50 mols?

Reply:
Yes good idea. Either a warning or make it clear in the docstring. Thanks!